fets/model/project/mlcube.py

```diff
@@ -1,59 +1,48 @@
 """MLCube handler file"""
 import typer
 import yaml
-from src.my_logic import run_code
-
-app = typer.Typer()
+from src.my_logic import run_inference
 
 
-class ExampleTask(object):
-    """Example task Class
-    It reads the content of the parameters file and then
-    prints "MY_NEW_PARAMETER_EXAMPLE"."""
-
-    @staticmethod
-    def run_example(parameters_file: str) -> None:
-
-        # Load parameters from the paramters file
-        with open(parameters_file, "r") as stream:
-            parameters = yaml.safe_load(stream)
-
-        print("This is my new parameter example:")
-        print(parameters["MY_NEW_PARAMETER_EXAMPLE"])
+# This is used to create a simple CLI``
+app = typer.Typer()
 
 
 class InferTask(object):
-    """Run task Class
-    It defines the environment variables:
+    """ Inference task
+    This class defines the environment variables:
         data_path: Directory path to dataset
         output_path: Directory path to final results
+        checkpoint_path: Directory path to model checkpoints
         All other parameters are defined in parameters_file
-    Then executes the run_code method inside my_logic script"""
+    The `run` method executes the run_inference method from the src.my_logic module"""
 
     @staticmethod
-    def run(data_path: str, output_path: str, parameters_file: str) -> None:
-
+    def run(data_path: str, output_path: str, parameters_file: str, checkpoint_path: str) -> None:
         # Load parameters from the paramters file
         with open(parameters_file, "r") as stream:
             parameters = yaml.safe_load(stream)
 
         application_name = parameters["APPLICATION_NAME"]
         application_version = parameters["APPLICATION_VERSION"]
-        run_code(data_path, output_path, application_name, application_version)
-
-
-@app.command("example")
```

Comment:
@davidjurado I noticed that when I have only one app.command in this file, this breaks the program because it complains that the extra parameter "infer" was passed. Do you know if there is another solution based on `typer` than just having a dummy "example" command here?

Reply:
This is a known issue from the typer package, the solution they provide doesn't work for us because it implies that we don't pass any command, therefore we won't be able to specify the task name, that's why we need to have at least two commands inside the **mlcube.py** file (doesn't mean that these two need to be defined in the mlcube.yaml file).